extension.js

```diff
@@ -294,3 +294,135 @@ const getParamsFromGithubRaw = (url) => {
+    let url = `https://api.github.com/search/repositories?q=${encodeURIComponent("topic:spicetify-extensions")}&per_page=${ITEMS_PER_REQUEST}`;
+    const allRepos = await fetch(url).then(res => res.json()).catch(() => []);
+    if (!allRepos.items) {
+        Spicetify.showNotification("Too Many Requests, Cool Down.");
+    }
+    const filteredResults = {
+        ...allRepos,
+        items: allRepos.items.filter(item => !BLACKLIST.includes(item.html_url)),
+    };
+    return filteredResults;
+}
+async function Blacklist() {
+    const url = "https://raw.githubusercontent.com/CharlieS1103/spicetify-marketplace/main/blacklist.json";
+    const jsonReturned = await fetch(url).then(res => res.json()).catch(() => { });
+    return jsonReturned.repos;
+}
+(async function initializePreload() {
+
+    // Begin by getting the themes and extensions from github
+    const extensionsArray = await storeExtensions();
+    const themesArray = await storeThemes();
+    window.sessionStorage.clear();
+    appendInformationToLocalStorage(themesArray, "theme");
+    appendInformationToLocalStorage(extensionsArray, "extension");
+})();
+
+async function appendInformationToLocalStorage(array, type) {
+    // This system should make it so themes and extensions are stored concurrently
+    if (type == "theme") {
+        for (const repo of array.items) {
+            let themes = await fetchThemes(repo.contents_url, repo.default_branch);
+            if (themes) {
+                addToSessionStorage(themes);
+                await sleep(10000);
+            }
+        }
+    } else if (type == "extension") {
+        for (const repo of array.items) {
+            let extensions = await fetchExtensions(repo.contents_url, repo.default_branch);
+            if (extensions) {
+                addToSessionStorage(extensions);
+                await sleep(10000);
+            }
+
+        }
+    }
+}
+// This function is used to fetch manifest of a theme and return it
+async function fetchThemes(contents_url, branch) {
+    try {
+        const regex_result = contents_url.match(/https:\/\/api\.github\.com\/repos\/(?<user>.+)\/(?<repo>.+)\/contents/);
+        // TODO: err handling?
+        if (!regex_result || !regex_result.groups) return null;
+        let { user, repo } = regex_result.groups;
+        let manifests = await getRepoManifest(user, repo, branch);
+        // If the manifest returned is not an array, initialize it as one
+        if (!Array.isArray(manifests)) manifests = [manifests];
+        manifests.user = user;
+        manifests.repo = repo;
+        if (manifests[0] && manifests[0].name && manifests[0].usercss && manifests[0].description) {
+            return manifests;
+        }
+        return null;
+    }
+    catch (err) {
+        // console.warn(contents_url, err);
+        return null;
+    }
+}
+// This function is used to fetch manifest of an extension and return it
+async function fetchExtensions(contents_url, branch) {
```

Comment:
Can we rename this to `fetchExtensionManifest` or something? The name is confusing with `storeExtensions`.